aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -584,8 +584,8 @@ std::tuple<Tensor, Tensor> mode(const Tensor& self, int64_t dim, bool keepdim) {
 
 std::tuple<Tensor &,Tensor &> mode_out(const Tensor& self, int64_t dim, bool keepdim,
                                        Tensor& values, Tensor& indices) {
-  TORCH_CHECK(self.device().is_cpu() || self.is_cuda(),
-              "mode only supports CPU AND CUDA device type, got: ", self.device().type());
+  TORCH_CHECK(self.device().is_cpu() || self.is_cuda() || self.is_xpu(),
```

Comment:
Regarding the device check, I'm wondering if we should add a tensor method (`is_accelerator`) to the tensor class. @albanD , thoughts?

Reply:
I think it's a good idea to generalize it